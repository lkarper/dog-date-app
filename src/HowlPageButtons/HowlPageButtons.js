import React, { useContext, useState } from 'react';
import UserContext from '../contexts/UserContext';
import HowlsService from '../services/howls-service';

const HowlPageButtons = (props) => {

    const { howl_id } = props;

    const context = useContext(UserContext);

    const [apiError, setApiError] = useState(false);

    const howlSaved = context.userSavedHowls.find(ush => ush.howl.id === howl_id);

    const removeHowlCheck = () => {
        const confirmation = window.confirm(`Are you sure that you'd like to remove this howl from your list of saved howls?`);
        if (confirmation) {
            setApiError(false);
            HowlsService.removeHowlFromUserSaved(howlSaved.id)
                .then(res => {
                    context.removeSavedHowl(howl_id);
                })
                .catch(error => {
                    console.log(error);
                    setApiError(true);
                });
        }
    }

    const addHowlCheck = () => {
        const confirmation = window.confirm(`Would you like to add this howl to your saved howls list?`);
        if (confirmation) {
            setApiError(false);
            const newSavedHowl = {
                howl_id
            };
            HowlsService.addUserSavedHowl(newSavedHowl)
                .then(howl => {
                    context.addSavedHowl(howl);
                })
                .catch(error => {
                    console.log(error);
                    setApiError(true);
                });
        }
    }

    if (howlSaved) {
        return (
            <div aria-live='polite'>
                <p>Howl saved!</p>
                <button onClick={removeHowlCheck}>Remove howl from your saved howls.</button>
                <div role='alert'>
                    {apiError && <p>Error: Looks like something went wrong. Please check your connection and try again.</p>}
                </div>
            </div>
        );
    }

    return (
        <div aria-live='polite'>
            <button onClick={addHowlCheck}>Save howl!</button>
            <div role='alert'>
                {apiError && <p>Error: Looks like something went wrong. Please check your connection and try again.</p>}
            </div>
        </div>
    );
}

export default HowlPageButtons;
