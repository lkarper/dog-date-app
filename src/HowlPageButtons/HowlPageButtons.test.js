import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import HowlPageButtons from './HowlPageButtons';

describe('HowlPageButtons component', () => {
    it('renders the UI without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<HowlPageButtons />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const wrapper = mount(<HowlPageButtons />);
        expect(toJSON(wrapper)).toMatchSnapshot();
        wrapper.unmount();
    });
});
