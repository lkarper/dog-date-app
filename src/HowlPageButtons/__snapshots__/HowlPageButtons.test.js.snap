// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HowlPageButtons component renders the UI as expected 1`] = `
<HowlPageButtons
  howl_id=""
>
  <div
    className="error"
    role="alert"
  >
    <p>
      Error: Looks like something went wrong. Please check your connection and the URL and try again.
    </p>
  </div>
</HowlPageButtons>
`;
